src/countBy.js

```diff
@@ -25,8 +24,12 @@ var _keyValue = require('./internal/_keyValue');
  *      R.countBy(R.toLower)(letters);   //=> {'a': 5, 'b': 4, 'c': 3}
  */
 module.exports = _curry2(function countBy(fn, list) {
-    return _foldl(function(counts, obj) {
-        counts[obj.key] = (counts[obj.key] || 0) + 1;
-        return counts;
-    }, {}, _keyValue(fn, list));
+    var counts = {};
+    var len = list.length;
+    var idx = -1;
+    while (++idx < len) {
+        var key = fn(list[idx]);
+        counts[key] = (has(key, counts) ? counts[key] : 0) + 1;
+    }
+    return counts;
 });
```

Comment:
Good idea. I've added a second commit to this pull request, which removes `_compareKeys` and `_keyValue`. :)